### ISSUE TITLE
[BUG] Overlapping ROI with finetuned Cellpose

### ISSUE THREAD

**Issue by postnubilaphoebus** [MouseLand/cellpose#890]
**Describe the bug**
Cellpose seems to produce overlapping ROIs for different cells.
In the paper it says "The Cell-
pose model cannot predict overlapping ROI". Does that apply only for the vanilla Cellpose or also when one finetunes a version of it?

**To Reproduce**
I ran a 3d tif stack with a finetuned version of cellpose in the provided GUI.
Below, I have a attached a screenshot of the overlapping ROI of three cells. 

**Screenshots**
![overlap_proof](https://github.com/MouseLand/cellpose/assets/63673809/762a3871-3338-4b80-8dbf-a9425b5eacaa)

**Comment by mrariden** [comment]
In 2D Cellpose will label a single pixel/voxel as belong to only a single cell. It is not capable of attributing a single pixel/voxel to multiple cells. This is what is meant by "non-overlapping". This constraint is built into the design of cellpose, regardless of the specific pretrained model. 

Cellpose 3D will run on a z-stack but really treats the images as 2d. There is a possibility that the z-direction post processing might produce weirdness. How does this data look in the cellpose gui?

**Comment by postnubilaphoebus** [comment]
![Screenshot from 2024-03-12 14-32-29](https://github.com/MouseLand/cellpose/assets/63673809/45300e87-f02a-4830-b8b4-a921b5fef783)
![Screenshot from 2024-03-12 14-32-10](https://github.com/MouseLand/cellpose/assets/63673809/bf97aa93-5fa4-4b02-a9f4-3143a9a56305)
![Screenshot from 2024-03-12 14-32-00](https://github.com/MouseLand/cellpose/assets/63673809/8115592f-22fc-42df-8c71-231406f42857)
![Screenshot from 2024-03-12 14-31-49](https://github.com/MouseLand/cellpose/assets/63673809/c235e033-1916-4bd5-ac6e-ff5037f7c3bb)
![adaptive_hist_equal](https://github.com/MouseLand/cellpose/assets/63673809/10d5d0b3-a16a-4815-a1f5-be3577ca073d)
Screenshots from the Cellpose GUI. Not the exact cell, but it brings the point across (I can look for it if that helps). In 3d, those cells will intertwine.

**Comment by mrariden** [comment]
It looks like your cell has disconnected components which make it appear 'inside' other cells. These cells look very small for cellpose, which is optimized to segment cells with a diameter of 30 pixels. I'd start with improving the segmentation (either using higher resolution images, or maybe using Cellpose3 'upsample') if this issue is persistent.

Still, and per your question, the 3d voxels will only belong to a single cell. A single voxel cannot belong to multiple cells. If this answers your question, please close the issue

**Comment by postnubilaphoebus** [comment]
Obtaining a higher resolution is unfortunately not possible. I will try a different approach by calculating some fraction between the surface area and the volume of the cells to find odd instances. The upsampling sounds interesting, but unfortunately is only available in 2D in the GUI. Is that intentional?

**Comment by mrariden** [comment]
Ah sorry, yes it is not implemented for 3d yet. It is only implemented for 2d.
